core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/GraphicElementFactory.java

```diff
@@ -187,15 +203,38 @@ public SettingBundle getFavoriteLookSettings() {
    */
   public final void setLook(String lookName) {
     // get the customer lookSettings
-    String selectedLook = lookSettings.getString(lookName, null);
-    if (selectedLook == null) {
-      // ce look n'existe plus, look par defaut
-      selectedLook = defaultLook;
+    String look = lookSettings.getString(lookName, null);
+    if (!isLookAllowed(lookName)) {
+      // this look no more exists, set an existing look
+      lookName = getAvailableLooksForUser().get(0);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "lookName". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1226)